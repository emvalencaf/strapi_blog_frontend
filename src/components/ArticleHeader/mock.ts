import { ArticleHeaderProps } from ".";

export default {
	title: "lorem ipsum title",
	excerpt: "lorem ipsum lorem impsum",
	cover: {
		url: "image.jpg",
	},
	createdAt: "2021-03-02T15:16:09.246+00:00",
	author: {
		id: "woek2131239481",
		slug: "lorem-ipsum",
		displayName: "Lorem ipsum",
	},
	categories: [
		{
			id: "ijmwerw19212",
			displayName: "Tech",
			slug: "tech",
		},
		{
			id: "wiejj293888923",
			displayName: "Javascript",
			slug: "javascript",
		},
	],
} as ArticleHeaderProps;
