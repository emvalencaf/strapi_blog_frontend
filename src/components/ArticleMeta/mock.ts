import { ArticleMetaProps } from ".";

export default {
	createdAt: "2021-03-02T15:16:09.246+00:00",
	author: {
		id: "woek2131239481",
		slug: "lorem-ipsum",
		displayName: "Lorem ipsum",
	},
	categories: [
		{
			id: "ijmwerw19212",
			displayName: "Tech",
			slug: "tech",
		},
		{
			id: "wiejj293888923",
			displayName: "Javascript",
			slug: "javascript",
		},
	],
} as ArticleMetaProps;
