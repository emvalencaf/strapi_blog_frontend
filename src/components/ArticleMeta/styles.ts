import styled, { css } from "styled-components";

export const Wrapper = styled.footer`
	${({ theme }) => css`
		font-size: ${theme.font.sizes.small};
		color: ${theme.colors.darkerGray};

		.categories span::after {
			content: ", ";
		}

		.categories span:last-child::after {
			content: "";
		}

		a {
			color: ${theme.colors.secondary};
			text-decoration: none;
			transition: : all 350ms ease-in-out;

			&:hover {
				filter: brightness(50%);
			}
		}


	`}
`;
