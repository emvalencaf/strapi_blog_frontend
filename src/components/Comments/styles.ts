import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${({ theme }) => css`
		max-width: ${theme.sizes.content};
		width: 100%;
		margin: ${theme.spacings.large} 0;
		padding: 0 ${theme.spacings.large};
	`}
`;
