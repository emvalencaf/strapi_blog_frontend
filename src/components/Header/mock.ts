import { HeaderProps } from ".";

export default {
	blogName: "lorem ipsum blog",
	blogDescription:
		"Lorem, ipsum dolor sit amet consectetur adipisicing elit. Consequatur, dolorum adipisci distinctio tenetur consequuntur numquam sed recusandae nulla quidem, quam necessitatibus enim culpa. At modi est reprehenderit odit quaerat ut.",
	logo: "teste.jpg",
} as HeaderProps;
