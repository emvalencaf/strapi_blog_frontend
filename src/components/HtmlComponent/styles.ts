import styled, { css } from "styled-components";

export const Container = styled.div`
	${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
      }
    }

    font-size: ${theme.font.sizes.medium};


    p {
      margin: ${theme.spacings.xlarge} 0;
    }

    a, a:visited, a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 350ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      font-size: ${theme.font.sizes.small};
      background-color: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white}
      margin: ${theme.spacings.xlarge};
      width: 100%;
      overflow-x: auto;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }


    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid $${theme.colors.mediumGray};
    }

    ul, ol {
      margin: ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;

    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    .table td, .table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
  `}
`;
