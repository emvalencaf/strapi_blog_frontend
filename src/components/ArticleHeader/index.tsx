// imported types
import { Author } from "../../shared-typed/author";
import { Category } from "../../shared-typed/category";
import { StrapImage } from "../../shared-typed/strap-image";
import ArticleMeta from "../ArticleMeta";
import Heading from "../Heading";

// styles
import * as Styled from "./styles";

// types
export type ArticleHeaderProps = {
	title: string;
	excerpt: string;
	createdAt: string;
	cover: StrapImage;
	author: Author;
	categories: Category[];
};

const ArticleHeader = ({
	title,
	excerpt,
	cover,
	createdAt,
	author,
	categories,
}: ArticleHeaderProps) => {
	return (
		<Styled.Wrapper>
			<Heading as="h2" size="medium" colorDark={true}>
				{title}
			</Heading>
			<Styled.Excerpt>{excerpt}</Styled.Excerpt>
			<Styled.Cover src={cover.url} alt={title} />
			<ArticleMeta
				createdAt={createdAt}
				author={author}
				categories={categories}
			/>
		</Styled.Wrapper>
	);
};

export default ArticleHeader;
