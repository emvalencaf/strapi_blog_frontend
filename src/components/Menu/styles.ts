import styled, { css, DefaultTheme } from "styled-components";

// imported styled
import { Title as HeadingStyles } from "../Heading/styles";

// types
type MenuBehaviorParams = {
	menuVisible: boolean;
	theme: DefaultTheme;
};

const wrapperChanger = (menuVisible: boolean, theme: DefaultTheme) => css`
	@media ${theme.media.lteMedium} {
		left: ${menuVisible ? "0" : "-32rem"};
	}

	left: ${menuVisible ? "0" : "-30rem"};
	overflow-y: ${menuVisible ? "auto" : "hidden"};
`;

const buttonChanger = (menuVisible: boolean, theme: DefaultTheme) => css`
	@media ${theme.media.lteMedium} {
		left: ${menuVisible ? "26rem" : "1rem"};
	}

	left: ${menuVisible ? "26rem" : "-1rem"};
	color: ${menuVisible ? theme.colors.secondary : theme.colors.white};
`;

export const Wrapper = styled.div<MenuBehaviorParams>`
	${({ theme, menuVisible }) => css`
		background: ${theme.colors.primary};
		padding: ${theme.spacings.large};
		display: flex;
		position: fixed;
		z-index: 1;
		width: 100%;
		max-width: 32rem;
		height: 100vh;
		top: 0;
		left: 0;
		transition: all 350ms ease-in-out;
		${wrapperChanger(menuVisible, theme)}
	`}
`;

export const Nav = styled.nav`
	${() => css`
		margin: auto;
		width: 100%;
	`}
`;

export const Logo = styled.div`
	${({ theme }) => css`
		${HeadingStyles} {
			display: flex;
			justify-content: center;
			margin: 0;
			margin-bottom: ${theme.spacings.xlarge};

			img {
				border: 0.5rem solid ${theme.colors.secondary};
			}
		}
	`}
`;

export const OpenClose = styled.a<MenuBehaviorParams>`
	${({ theme, menuVisible }) => css`
		position: fixed;
		top: ${theme.spacings.medium};
		color: ${theme.colors.white};
		background: ${theme.colors.primary};
		z-index: 2;
		width: 3rem;
		height: 3rem;
		left: 26rem;
		transition: all 350ms ease-in-out;
		${buttonChanger(menuVisible, theme)}
	`}
`;
