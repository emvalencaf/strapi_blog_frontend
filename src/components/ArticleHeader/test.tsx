import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";

// react component to be test
import ArticleHeader from ".";

// mock
import mock from "./mock";

describe("<ArticleHeader />", () => {
	it("should render ArticleHeader component with image, heading and ArticleMeta", () => {
		renderTheme(<ArticleHeader {...mock} />);

		expect(
			screen.getByRole("heading", { name: mock.title })
		).toBeInTheDocument();
		expect(screen.getByAltText(mock.title)).toHaveAttribute(
			"src",
			mock.cover.url
		);
		expect(
			screen.getByRole("link", { name: mock.author.displayName })
		).toHaveAttribute("href", `/author/${mock.author.slug}`);
	});

	it("should match snapshot", () => {
		const { container } = renderTheme(<ArticleHeader {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
