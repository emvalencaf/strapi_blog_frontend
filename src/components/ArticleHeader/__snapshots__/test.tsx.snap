// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleHeader /> should match snapshot 1`] = `
.c5 {
  font-size: 1.6rem;
  color: #AAAAAA;
}

.c5 .categories span::after {
  content: ", ";
}

.c5 .categories span:last-child::after {
  content: "";
}

.c5 a {
  color: #dc143c;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: :all 350ms ease-in-out;
  transition: :all 350ms ease-in-out;
}

.c5 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c2 {
  color: #000000;
  font-size: 3.2rem;
  text-transform: none;
}

.c0 {
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0.c1 {
  margin: 0;
  margin-bottom: 2.4rem;
}

.c3 {
  margin: 2.4rem 0;
}

.c4 {
  max-width: 100%;
  display: block;
  margin-bottom: 2.4rem;
}

<div>
  <header
    class="c0"
  >
    <h2
      class="c1 c2"
    >
      lorem ipsum title
    </h2>
    <p
      class="c3"
    >
      lorem ipsum lorem impsum
    </p>
    <img
      alt="lorem ipsum title"
      class="c4"
      src="image.jpg"
    />
    <footer
      class="c5"
    >
      <p>
        <span>
          Por 
        </span>
        <a
          href="/author/lorem-ipsum"
        >
          Lorem ipsum
        </a>
        <span
          class="separator"
        >
           | 
        </span>
        <time
          datetime="2021-03-02T15:16:09.246+00:00"
        >
          2 de mar. de 2021
        </time>
        <span
          class="separator"
        >
           | 
        </span>
        <span
          class="categories"
        >
          <span>
            <a
              href="/category/tech"
            >
              Tech
            </a>
          </span>
          <span>
            <a
              href="/category/javascript"
            >
              Javascript
            </a>
          </span>
        </span>
      </p>
    </footer>
  </header>
</div>
`;
